packages/expo/src/notifier.js

```diff
@@ -26,49 +31,74 @@ const plugins = [
 
 const bugsnagReact = require('@bugsnag/plugin-react')
 
-module.exports = (opts) => {
-  // handle very simple use case where user supplies just the api key as a string
-  if (typeof opts === 'string') opts = { apiKey: opts }
-
-  // ensure opts is actually an object (at this point it
-  // could be null, undefined, a number, a boolean etc.)
-  opts = { ...opts }
-
-  // attempt to fetch apiKey from app.json if we didn't get one explicitly passed
-  if (!opts.apiKey &&
-    Constants.manifest &&
-    Constants.manifest.extra &&
-    Constants.manifest.extra.bugsnag &&
-    Constants.manifest.extra.bugsnag.apiKey) {
-    opts.apiKey = Constants.manifest.extra.bugsnag.apiKey
-  }
+module.exports.default = module.exports
 
-  const bugsnag = new Client({ name, version, url })
-
-  bugsnag.delivery(delivery)
-  bugsnag.setOptions(opts)
-  bugsnag.configure(schema)
-
-  plugins.forEach(pl => {
-    switch (pl.name) {
-      case 'networkBreadcrumbs':
-        bugsnag.use(pl, () => [
-          bugsnag.config.endpoints.notify,
-          bugsnag.config.endpoints.sessions,
-          Constants.manifest.logUrl
-        ])
-        break
-      default:
-        bugsnag.use(pl)
+const Bugsnag = {
+  _client: null,
+  createClient: (opts) => {
+    // handle very simple use case where user supplies just the api key as a string
+    if (typeof opts === 'string') opts = { apiKey: opts }
+    if (!opts) opts = {}
+
+    // attempt to fetch apiKey from app.json if we didn't get one explicitly passed
+    if (!opts.apiKey &&
+      Constants.manifest &&
+      Constants.manifest.extra &&
+      Constants.manifest.extra.bugsnag &&
+      Constants.manifest.extra.bugsnag.apiKey) {
+      opts.apiKey = Constants.manifest.extra.bugsnag.apiKey
     }
-  })
-  bugsnag.use(bugsnagReact, React)
 
-  bugsnag._logger.debug(`Loaded!`)
+    const bugsnag = new Client(opts, schema, { name, version, url })
+
+    bugsnag._delivery(delivery)
+
+    plugins.forEach(pl => {
+      switch (pl.name) {
+        case 'networkBreadcrumbs':
+          bugsnag.use(pl, () => [
+            bugsnag._config.endpoints.notify,
+            bugsnag._config.endpoints.sessions,
+            Constants.manifest.logUrl
+          ])
+          break
+        default:
+          bugsnag.use(pl)
+      }
+    })
+    bugsnag.use(bugsnagReact, React)
+
+    bugsnag.__logger.debug('Loaded!')
+
+    if (bugsnag._config.autoTrackSessions) bugsnag.startSession()
 
-  return bugsnag.config.autoCaptureSessions
-    ? bugsnag.startSession()
-    : bugsnag
+    return bugsnag
+  },
+  init: (opts) => {
+    if (Bugsnag._client) {
+      Bugsnag._client.__logger.warn('init() called twice')
```

Comment:
Let's discuss what would be a good message for this warning and consider adding it to the spec. Perhaps this will kick off the speccing of further warnings and messages so we acheive consistency in that regard too?